### PR TITLE
CreateToken | Auth | Encrypt | Decrypt for JWT (Solves #1)

### PR DESCRIPTION
**Pull request by whyDontI**
Added following methods for JWT tokens 
CreateToken | Auth | Encrypt | Decrypt.
Which solves #1

### CODE REVIEW

File: src/middleware/auth/auth.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        req.userId = verifyUser.userId
+        next()
+      };
+    } catch (error) {
+      return __.errorMsg(req, res, 401, error.message, error, 'authentication')
+    }
+  }
+};
+
+module.exports = new Jwt()
```
Comment (Swap76) [inline src/middleware/auth/auth.js]: Hey @whyDontI ,

Can you please explain to me how this ```js new Jwt()``` will be used in different controllers

Comment (whyDontI) [reply src/middleware/auth/auth.js]: Hey @Swap76, we can simply use methods written in auth.js in other files by simply importing this file as ```const jwt = require('middleware/Auth/auth')```
Then, if I have to create JWT token I'll just write,
```const token = await jwt.createToken(user._id)```

Comment (whyDontI) [reply src/middleware/auth/auth.js]: I'll use ```authentication()``` method to authenticate any request being received by the server.
I'll just have to pass it as an argument like below, 
```app.get('/', auth.authentication, UserService._getAllUsers) ```

**Comment by whyDontI** [comment]
Thanks for the merge @Swap76 :)

File: src/middleware/auth/auth.js (hunk 1 above)
Comment (Swap76) [reply src/middleware/auth/auth.js]: Cool good implementation of classes

**Comment by Swap76** [comment]
@whyDontI  It was great working with you.

Looking forward to working with you :-)

Star to the repo is highly appreciated.

**Comment by Swap76** [comment]
@whyDontI The work you have done for the project is highly appreciated. Thanks :-)

Looking forward to working with you on my new project related to Deno https://github.com/Swap76/Learn-Deno/ and together making this better.

Star to the repo is highly appreciated.

**Comment by whyDontI** [comment]
Sure @Swap76 ,
It'll be a great chance for me to get started with deno.
Thanks.

**Pull request merged**